packages/simcore-sdk/src/simcore_sdk/node_ports_v2/ports_mapping.py

```diff
@@ -0,0 +1,44 @@
+from typing import Dict, ItemsView, Iterator, KeysView, Union, ValuesView
+
+from models_library.services import PROPERTY_KEY_RE
+from pydantic import BaseModel, constr
+
+from ..node_ports.exceptions import UnboundPortError
+from .port import Port
+
+PortKey: constr = constr(regex=PROPERTY_KEY_RE)
+
+
+class PortsMapping(BaseModel):
+    __root__: Dict[PortKey, Port]
+
+    def __getitem__(self, key: Union[int, PortKey]) -> Port:
+        if isinstance(key, int):
+            if key < len(self.__root__):
+                key = list(self.__root__.keys())[key]
+        if not key in self.__root__:
+            raise UnboundPortError(key)
+        return self.__root__[key]
+
+    def __iter__(self) -> Iterator[PortKey]:
```

Comment:
I would have expected that pydantic would be doing this automatically from the fact that you told that __root__ is Dict. This explains the issues I had with the workbench when I defined it this way instead of creating an alias of a ``Dict[str,...]``

Reply:
yes it is a bit painful. 
quoted from pydantic docs:
```
f you want to access items in the __root__ field directly or to iterate over the items, you can implement custom __iter__ and __getitem__ functions, as shown in the following example.
```

Reply:
also to have to use __root__ is not nice